src/Database/SqlDialectTrait.php

```diff
@@ -206,6 +206,14 @@ protected function _updateQueryTranslator($query)
      */
     protected function _removeAliasesFromConditions($query)
     {
+        if (!empty($query->clause('join'))) {
+            trigger_error(
```

Comment:
@ADmad yes, I understand your concern, but this is an edge case that won't happen. The ORM currently does not support JOINs on UPDATE/DELETE. This error is to future proof changes so that when they are added this code is updated.